sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -217,22 +320,153 @@ private void validateDocCRUDandQuery() throws Exception {
             //Run the query.
             String query = String.format("SELECT * from c where c.id = '%s'", doc_current.get("id").asText());
 
-            CosmosPagedIterable<ObjectNode> feedResponseIterator1 =
+            CosmosPagedIterable<ObjectNode> feedResponseIterator =
                 createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
-            assertThat(feedResponseIterator1.iterator().hasNext()).isTrue();
+            assertThat(feedResponseIterator.iterator().hasNext()).isTrue();
 
             query = String.format("SELECT * from c where c.id = '%s'", doc_current.get("id").asText());
             queryRequestOptions = new CosmosQueryRequestOptions();
-            feedResponseIterator1 =
+            feedResponseIterator =
                 createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
-            assertThat(feedResponseIterator1.iterator().hasNext()).isTrue();
+            assertThat(feedResponseIterator.iterator().hasNext()).isTrue();
         }
 
-        String query = String.format("SELECT * from c where c.city = '%s'", docs.get(2).get("city").asText());
+        partitionKey = new PartitionKeyBuilder()
+            .add(doc5.get("city").asText())
+            .add(doc5.get("zipcode").asText())
+            .build();
+
+        String query = "SELECT * from c";
         CosmosQueryRequestOptions queryRequestOptions = new CosmosQueryRequestOptions();
-        CosmosPagedIterable<ObjectNode> feedResponseIterator1 =
+        queryRequestOptions.setPartitionKey(partitionKey);
+        CosmosPagedIterable<ObjectNode> feedResponseIterator =
             createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
-        assertThat(feedResponseIterator1.stream().count()).isEqualTo(3);
+        assertThat(feedResponseIterator.stream().count()).isEqualTo(2);
+
+        query = String.format("SELECT * from c where c.city = '%s'", docs.get(2).get("city").asText());
+        queryRequestOptions = new CosmosQueryRequestOptions();
+        feedResponseIterator = createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
+        assertThat(feedResponseIterator.stream().count()).isEqualTo(3);
+
+        query = String.format("SELECT * from c where c.city = '%s'", docs.get(0).get("city").asText());
```

Comment:
Can we move all the query related tests into its own?since it has so many different test cases, might easier for reference in future